docs/tutorials/chatbot.md

```diff
@@ -0,0 +1,728 @@
+            print(f'foo is doing cool stuff: {kwargs}')
+    class MyIndexer(Executor):
+        @requests(on='/bar')
+        def bar(self, **kwargs):
+            print(f'bar is doing cool stuff: {kwargs}')
+    targets = {
+        'covid-csv': {
+            'url': args.index_data_url,
+            'filename': os.path.join(args.workdir, 'dataset.csv'),
+        }
+    }
+    # download the data
+    download_data(targets, args.download_proxy, task_name='download covid-dataset')
+    flow = (
+        Flow(cors=True)
+            .add(name='MyTransformer', uses=MyTransformer)
+            .add(name='MyIndexer', uses=MyIndexer)
+            .plot('test.svg')
+    )
+    with flow, open(targets['covid-csv']['filename']) as fp:
+        flow.index(from_csv(fp, field_resolver={'question': 'text'}))
+    
+        # switches the serving protocol to HTTP at runtime
+        flow.protocol = 'http'
+        flow.port_expose = args.port_expose
+        url_html_path = 'file://' + os.path.abspath(
+            os.path.join(
+                os.path.dirname(os.path.realpath(__file__)), 'static/index.html'
+            )
+        )
+        try:
+            webbrowser.open(url_html_path, new=2)
+        except:
+            pass  # intentional pass, browser support isn't cross-platform
+        finally:
+            default_logger.success(
+                f'You should see a demo page opened in your browser, '
+                f'if not, you may open {url_html_path} manually'
+            )
+        if not args.unblock_query_flow:
+            flow.block()
+```
+
+```{admonition} See Also
+:class: seealso
+For more information on what the Flow is doing, and how to serve the flow with `f.block()` and configure the protocol, 
+check {ref}`the Flow fundamentals section <flow-cookbook>`.
+```
+
+```{admonition} Important
+:class: important
+Since we want to call our Flow from the browser, it's important to enable 
+[Cross-Origin Resource Sharing](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) with `Flow(cors=True)`
+```
+
+
+
+Ok, so it seems that we have plenty of work done already. If you run this you will see a new tab open in your browser, 
+and there you will have a text box ready for you to input some text. However, if you try to enter anything you won't 
+get any results. This is because we are using dummy Executors. Our `MyTransformer` and `MyIndexer` aren't actually 
+doing anything. So far they only print a line when they are called. So we need real Executors.
+
+## Creating Executors
+We will be creating our Executors in a separate file: `my_executors.py`.
+
+First, let's import the following:
+```python
+from typing import Optional, Dict
+
+import numpy as np
+import torch
+from jina import Executor, DocumentArray, requests
+from jina.types.arrays.memmap import DocumentArrayMemmap
+from transformers import AutoModel, AutoTokenizer
+```
+
+Now, let's implement `MyTransformer`:
+```python
+class MyTransformer(Executor):
```

Comment:
Use the `sentence-encoders` package, this will drastically reduce lines of code. And keep in mind this is a tutorial - it is fine to set some defaults and not offer customization through params.

Reply:
* `sentence-transformers`

Reply:
done